admin/src/views/Analytics.vue

```diff
@@ -0,0 +1,293 @@
+    <BarChart class="marginClass" :chartData="chartDataBarFallback" :options="chartOptions" />
+    <LineChart class="marginClass" :chartData="chartDataLineIntents"
+      :options="chartDataLineIntentsOptions"  />
+    <el-select v-model="doghnutPeriod"
+      placeholder="Select Period" @change="selectPeriodForDoghnut">
+    <el-option
+      v-for="item in ['Last day', 'Last 7 days', 'Last 30 days']"
+      :key="item.value"
+      :label="item"
+      :value="item">
+    </el-option>
+  </el-select>
+    <DoghnutChart class="marginClass" :chartData="chartDoghnutIntent" :options="doghnutOptions"/>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import LineChart from '@/components/analytics/LineChart.vue';
+import BarChart from '@/components/analytics/BarChart.vue';
+import DoghnutChart from '@/components/analytics/DoghnutChart.vue';
+import dayjs from 'dayjs';
+import _ from 'lodash';
+import { colorsArray } from '@/utils';
+import { getAnalytics } from '@/client/analytics';
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+export default Vue.extend({
+  name: 'analytics',
+  components: {
+    LineChart,
+    BarChart,
+    DoghnutChart,
+  },
+  data() {
+    const handleIntentLegendClick = function (this: any, e, legendItem) {
+      const index = legendItem.datasetIndex;
+      const ci = this.chart;
+      const size = ci.data.datasets.length;
+      for (let i = 0; i < size; i += 1) {
+        if (i !== index) {
+          const meta = ci.getDatasetMeta(i);
+          meta.hidden = meta.hidden === null ? !ci.data.datasets[i].hidden : null;
+        }
+      }
+      // rerender the chart
+      ci.update();
+    };
+    return {
+      agentName: this.$route.params.agentName,
+      textMsg: '',
+      classMsg: '',
+      analyticsData: '',
+      period: 'Last 30 days',
+      doghnutPeriod: 'Last day',
+      chartOptions: {
+        responsive: true,
+        maintainAspectRatio: false,
+        scales: {
+          yAxes: [{
+            ticks: {
+              beginAtZero: true,
+            },
+          }],
+        },
+      },
+      chartDataLineIntentsOptions: {
+        responsive: true,
+        maintainAspectRatio: false,
+        scales: {
+          yAxes: [{
+            display: false,
+            ticks: {
+              stepSize: 1,
+              beginAtZero: true,
+            },
+          }],
+        },
+        legend: {
+          onClick: handleIntentLegendClick,
+        },
+        title: {
+          display: true,
+          text: 'Intents',
+          fontSize: 15,
+          fontStyle: 'normal',
+        },
+      },
+      doghnutOptions: {
+        responsive: true,
+        maintainAspectRatio: false,
+        legend: {
+          position: 'right',
+        },
+        title: {
+          display: true,
+          text: 'Intents Doghnut',
+          fontSize: 15,
+          fontStyle: 'normal',
+        },
+      },
+      dateFields: '',
```

Comment:
use Data or DataList (or another thing if you have suggestion) instead of Fields for all variables